### DISCUSSION TITLE
DEV Question. Local Server , how to debug a custom node w/o restarting server.

### DISCUSSION THREAD

**Discussion by Maelstrom2014** [comfyanonymous/ComfyUI#748] [General]
Hello, Everyone!
I'm new here. If i write in wrong tree, please point me were to ask dev questions, thanx.

So, I'm make a custom node and making constant changes to script. 

**Is it any way to re-run just edited (.py) custom script, W/O restarting server?** 

It take too much time even locally.

I tried to delete pychon cache, but no luck.
Refreshing a page, not helps.

**Comment by tusharbhutt** [comment]
I would also like to know the answer from some of the senior developers :)

**Comment by MaraScott** [comment]
Same

**Comment by jn-jairo** [comment]
For me restarting the server is pretty fast, because after the first run all python files stay cached in the RAM by the OS, so just a few seconds, most of the time it is like 5-6 seconds to restart, but it would be nice to have a faster way of doing it anyway.

Btw, I use linux ubuntu, I don't know if other OS works in that same way regarding file caching, for me even the safetensor is cached by the OS and load fast after the restart.

**Comment by mike-rowley** [comment]
Sorry to necro an old thread, but in case anyone else comes across this thread, here is a decent guide on getting debugging running.  Method 1 with a portable comfyui works well and restarts quickly, especially if you add the restart button to the menu in the second link.

https://civitai.com/articles/3654/a-programmers-guide-on-debugging-comfyui-workflow

https://www.reddit.com/r/comfyui/comments/18tvhqc/is_there_a_restart_server_shortcut/

Debugging python works well, I haven't figured out JavaScript debugging however.
